api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1511,6 +1531,208 @@ public void getObject(String bucketName, String objectName, String fileName)
+   * decryption, the CipherInputStream will attempt to read in data and decrypt them, before returning the decrypted
+   * data.
+   * <p>
+   * The CipherInputStream must be closed after use else the connection will remain open.
+   * </p>
+   * <b>Example:</b>
+   *
+   * <pre>
+   * {
+   *
+   *   // Get object with symmetric master key used in putObject
+   *   InputStream stream = minioClient.getObject("my-bucketname", "my-objectname", key);
+   *   byte[] buf = new byte[16384];
+   *   int bytesRead;
+   *   while ((bytesRead = stream.read(buf, 0, buf.length)) >= 0) {
+   *     System.out.println(new String(buf, 0, bytesRead));
+   *   }
+   *   stream.close();
+   * }
+   * </pre>
+   *
+   * @param bucketName
+   *          Bucket name.
+   * @param objectName
+   *          Object name in the bucket.
+   * @param key
+   *          Symmetric key used for corresponding putOject operation.
+   *
+   * @return {@link InputStream} containing the object data.
+   *
+   * @throws InvalidBucketNameException
+   *           upon invalid bucket name is given
+   * @throws NoResponseException
+   *           upon no response from server
+   * @throws IOException
+   *           upon connection error
+   * @throws XmlPullParserException
+   *           upon parsing response xml
+   * @throws ErrorResponseException
+   *           upon unsuccessful execution
+   * @throws InternalException
+   *           upon internal library error
+   * @throws InvalidEncryptionMetadataException
+   *           upon encryption key/iv error
+   * @throws BadPaddingException
+   *           upon wrong padding
+   * @throws IllegalBlockSizeException
+   *           upon incorrect block size
+   * @throws NoSuchPaddingException
+   *           upon incorrect padding
+   * @throws InvalidAlgorithmParameterException
+   *           upon incorrect algorithm
+   * 
+   * @see #getObject(String bucketName, String objectName)
+   * @see #getObject(String bucketName, String objectName, long offset)
+   * @see #getObject(String bucketName, String objectName, long offset, Long length)
+   * @see #getObject(String bucketName, String objectName, String filename)
+   * @see #getObject(String bucketName, String objectName, KeyPair key)
+   * 
+   */
+  public InputStream getObject(String bucketName, String objectName, SecretKey key)
+      throws InvalidBucketNameException, NoSuchAlgorithmException, InsufficientDataException, IOException,
+      InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException, InternalException,
+      InvalidArgumentException, NoSuchPaddingException, IllegalBlockSizeException,
+      BadPaddingException, InvalidAlgorithmParameterException, InvalidEncryptionMetadataException {
+
+    if (key == null) {
+      throw new InvalidArgumentException("empty decryption key not allowed");
+    }
+    String keyDecryptionCipherMode = "AES/ECB/PKCS5Padding";
```

Comment:
Then how do you propose sending this info down to the private method? 

We need this variable to pass the transformation to be used for key encryption in the private function.